### ISSUE TITLE
OpenSSL Failure trying to build Standalone Application

### ISSUE THREAD

**Issue by tes-hid** [openweave/openweave-core#180]
I am building on Ubuntu 18.0.4.
I ran into the same issues building on the Windows Subsystem for Linux, which btw, I generally ignored. I know you all do not really support it. Anyway, the error is basically that I'm missing OpenSSL which is not the case. I have installed libssl-dev. I also made sure to install Text::Template from cpan.

```
In file included from ../../src/include/Weave/Support/crypto/WeaveCrypto.h:88:0,
                 from ../../src/include/Weave/Support/crypto/HashAlgos.h:46,
                 from ../../src/include/Weave/Profiles/security/WeavePASE.h:42,
                 from ../../src/include/Weave/Core/WeaveSecurityMgr.h:36,
                 from ../../src/include/Weave/Core/WeaveCore.h:61,
                 from ../../src/include/Warm/Warm.h:80,
                 from /home/pacs/weave/openweave-core/src/warm/../../src/warm/WarmCore.cpp:26:
../../src/include/Weave/Support/crypto/WeaveRNG.h:33:2: error: #error "INVALID WEAVE CONFIG: OpenSSL RNG implementation enabled but OpenSSL not available (WEAVE_CONFIG_RNG_IMPLEMENTATION_OPENSSL == 1 && WEAVE_WITH_OPENSSL == 0)."
 #error "INVALID WEAVE CONFIG: OpenSSL RNG implementation enabled but OpenSSL not available (WEAVE_CONFIG_RNG_IMPLEMENTATION_OPENSSL == 1 && WEAVE_WITH_OPENSSL == 0)."
  ^~~~~
In file included from ../../src/include/Weave/Support/crypto/WeaveCrypto.h:89:0,
                 from ../../src/include/Weave/Support/crypto/HashAlgos.h:46,
                 from ../../src/include/Weave/Profiles/security/WeavePASE.h:42,
                 from ../../src/include/Weave/Core/WeaveSecurityMgr.h:36,
                 from ../../src/include/Weave/Core/WeaveCore.h:61,
                 from ../../src/include/Warm/Warm.h:80,
                 from /home/pacs/weave/openweave-core/src/warm/../../src/warm/WarmCore.cpp:26:
../../src/include/Weave/Support/crypto/AESBlockCipher.h:35:2: error: #error "INVALID WEAVE CONFIG: OpenSSL AES implementation enabled but OpenSSL not available (WEAVE_CONFIG_AES_IMPLEMENTATION_OPENSSL == 1 && WEAVE_WITH_OPENSSL == 0)."
 #error "INVALID WEAVE CONFIG: OpenSSL AES implementation enabled but OpenSSL not available (WEAVE_CONFIG_AES_IMPLEMENTATION_OPENSSL == 1 && WEAVE_WITH_OPENSSL == 0)."
  ^~~~~
In file included from ../../src/include/Weave/Profiles/security/WeavePASE.h:42:0,
                 from ../../src/include/Weave/Core/WeaveSecurityMgr.h:36,
                 from ../../src/include/Weave/Core/WeaveCore.h:61,
                 from ../../src/include/Warm/Warm.h:80,
                 from /home/pacs/weave/openweave-core/src/warm/../../src/warm/WarmCore.cpp:26:
../../src/include/Weave/Support/crypto/HashAlgos.h:49:2: error: #error "INVALID WEAVE CONFIG: OpenSSL hash implementation enabled but OpenSSL not available (WEAVE_CONFIG_HASH_IMPLEMENTATION_OPENSSL == 1 && WEAVE_WITH_OPENSSL == 0)."
 #error "INVALID WEAVE CONFIG: OpenSSL hash implementation enabled but OpenSSL not available (WEAVE_CONFIG_HASH_IMPLEMENTATION_OPENSSL == 1 && WEAVE_WITH_OPENSSL == 0)."
  ^~~~~
In file included from ../../src/include/Weave/Core/WeaveSecurityMgr.h:36:0,
                 from ../../src/include/Weave/Core/WeaveCore.h:61,
                 from ../../src/include/Warm/Warm.h:80,
                 from /home/pacs/weave/openweave-core/src/warm/../../src/warm/WarmCore.cpp:26:
../../src/include/Weave/Profiles/security/WeavePASE.h:47:2: error: #error "INVALID WEAVE CONFIG: PASE Config1 enabled but OpenSSL not available (WEAVE_CONFIG_SUPPORT_PASE_CONFIG1 == 1 && WEAVE_WITH_OPENSSL == 0)."
 #error "INVALID WEAVE CONFIG: PASE Config1 enabled but OpenSSL not available (WEAVE_CONFIG_SUPPORT_PASE_CONFIG1 == 1 && WEAVE_WITH_OPENSSL == 0)."
  ^~~~~
In file included from ../../src/include/Weave/Profiles/security/WeavePASE.h:51:0,
                 from ../../src/include/Weave/Core/WeaveSecurityMgr.h:36,
                 from ../../src/include/Weave/Core/WeaveCore.h:61,
                 from ../../src/include/Warm/Warm.h:80,
                 from /home/pacs/weave/openweave-core/src/warm/../../src/warm/WarmCore.cpp:26:
../../src/include/Weave/Support/crypto/EllipticCurve.h:36:2: error: #error "INVALID WEAVE CONFIG: OpenSSL ECC implementation enabled but OpenSSL not available (WEAVE_CONFIG_USE_OPENSSL_ECC == 1 && WEAVE_WITH_OPENSSL == 0)."
 #error "INVALID WEAVE CONFIG: OpenSSL ECC implementation enabled but OpenSSL not available (WEAVE_CONFIG_USE_OPENSSL_ECC == 1 && WEAVE_WITH_OPENSSL == 0)."
```

Also attaching the config.log which I read in another issue might be helpful as well.

config.log: https://pastebin.com/GwvZtAMX

**Comment by jaylogue** [comment]
It's strange that this isn't working for you, as I develop on a 18.04 system as well.  

It would appear that configure is not finding the OpenSSL header files.  If you could provide some addition information it might be helpful.

1) Verify that the file /usr/include/openssl/aes.h exists on your system.

2) Cat the contents of /usr/lib/x86_64-linux-gnu/pkgconfig/openssl.pc and attach it here.

If either of those files are missing, then perhaps libssl-dev didn't get installed correctly on your system.

**Comment by tes-hid** [comment]
aes.h does exist.

Contents of openssl.pc
```
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib/x86_64-linux-gnu
includedir=${prefix}/include

Name: OpenSSL
Description: Secure Sockets Layer and cryptography libraries and tools
Version: 1.1.0g
Requires: libssl libcrypto
```

**Comment by tes-hid** [comment]
ok -- so I was just able to build. Just going to list a few things I have done -- not sure what actually helped really, but just for your info.

Set JAVA_HOME -- planning to experiment with the Android SDK and it wasn't set

`make distclean`
`./configure`
`make all`

That's all, but it seemed to work... so good news? 👍
